src/nni_manager/common/log.ts

```diff
@@ -134,14 +137,21 @@ class Logger {
     public fatal(...param: any[]): void {
         this.log('FATAL', param);
     }
-
+    
+    /**
+     * if the experiment is not in readonly mode, write log content to stream
+     * @param level log level
+     * @param param the params to be written
+     */
     private log(level: string, param: any[]): void {
-        const buffer: WritableStreamBuffer = new WritableStreamBuffer();
-        buffer.write(`[${(new Date()).toLocaleString()}] ${level} `);
-        buffer.write(format(param));
-        buffer.write('\n');
-        buffer.end();
-        this.bufferSerialEmitter.feed(buffer.getContents());
+        if (!this.readonly) {
+            const buffer: WritableStreamBuffer = new WritableStreamBuffer();
+            buffer.write(`[${(new Date()).toLocaleString()}] ${level} `);
+            buffer.write(format(param));
+            buffer.write('\n');
+            buffer.end();
+            this.bufferSerialEmitter.feed(buffer.getContents());
```

Comment:
Has the dispatcher log been handled?

Reply:
In readonly mode, the dispatcher process will not started.